src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectContextFactory.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal interface IProjectContextFactory
+    {
+        IProjectContext CreateProjectContext(string languageName, string projectDisplayName, string projectFilePath, Guid projectGuid, string projectTypeGuid, object hostObject, CommandLineArguments commandLineArguments);
```

Comment:
We are surely not going to have project type guids in project file, but I am not sure if they are not there even in memory/sln file - @davkean can probably answer that.